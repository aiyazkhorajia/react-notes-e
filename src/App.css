@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;500;700;900&display=swap');

*, body {
    font-family: 'Poppins', sans-serif;
    font-weight: 400;
    -webkit-font-smoothing: antialiased;
    text-rendering: optimizeLegibility;
    -moz-osx-font-smoothing: grayscale;
}

html, body {
    background-color: #152733;
    height: vh-100;
}

.form-container {
    width: 100%;
    max-width: 500px;
    margin: 0 auto;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.notes-container {
    color: #000;
}

.list-group-item {
    cursor: pointer;
}