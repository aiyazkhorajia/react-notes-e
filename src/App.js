import React, { Suspense } from 'react'
import { BrowserRouter as Router, Switch } from 'react-router-dom'
import PrivateRoute from 'Components/PrivateRoute'
import './App.css'

const Notes = React.lazy(() => import('Pages/Notes'))
const SignIn = React.lazy(() => import('Pages/SignIn'))

export default function App() {
  return (
    <Router>
      <Suspense fallback={<div>Loading...</div>}>
        <Switch>
          <PrivateRoute component={SignIn} path='/' exact type='public' />
          <PrivateRoute component={SignIn} path='/signin' type='public' />
          <PrivateRoute component={Notes} path='/notes' type='protected' />
        </Switch>{' '}
      </Suspense>
    </Router>
  )
}
