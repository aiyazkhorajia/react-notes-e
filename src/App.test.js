import React from 'react'
import App from './App'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { Provider } from 'react-redux'

describe('<App />', () => {
  let store, wrapper

  beforeEach(() => {
    store = configureStore([thunk])({
      signInData: {
        loading: false,
        error: null,
        isAuthenticated: false,
        user: null
      }
    })

    wrapper = shallow(
      <Provider store={store}>
        <App />
      </Provider>
    )
  })

  describe('on mount', () => {
    it('should renders correctly', () => {
      expect(wrapper.exists()).toBe(true)
    })
  })
})
