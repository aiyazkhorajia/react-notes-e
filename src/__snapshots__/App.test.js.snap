// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> on mount renders correctly 1`] = `
Array [
  <div
    className="form-body"
  >
    <div
      className="row"
    >
      <div
        className="form-holder"
      >
        <div
          className="form-content"
        >
          <div
            className="form-items"
          >
            <h3>
              Opps!
            </h3>
            <p>
              Something went wrong
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  " ",
]
`;
